core/state/state_object.go

```diff
@@ -128,11 +119,6 @@ func newObject(db *StateDB, address common.Address, data Account) *stateObject {
 	}
 }
 
-// EncodeRLP implements rlp.Encoder.
-func (s *stateObject) EncodeRLP(w io.Writer) error {
-	return rlp.Encode(w, s.data)
-}
```

Comment:
Before you remove it completely, might be better to place a `panic` in it, to make sure we never ever use it aside from the locations where you've alredy replaced it with `rlp.Encode(object.data)`